### PR TITLE
Fix firebase-storage UploadTest flakes

### PR DESCRIPTION
**Pull request by daymxn**
Per b/268217935,

This fixes the failing `UploadTest` test in `firebase-storage`. It seems the issue originated from a couple different places:
1.) `ControllableSchedulerHelper` wasn't correctly synchronizing when validating threads
2.) Previous test cases that have no use now were present
3.) Since `addOnFailureListener` calls don't originate from the same thread, it seems the `Assert` calls were throwing hidden exceptions

I've removed the message check, since it's sort of redundant. But if we chose to add it again, it should be added outside the scope of the failure listener.

This bug was particularly difficult to debug, as it only seemed to fail on CI. My assumption is that it has to do with how resources and threads are handled in the image we have on CI- but I couldn't find anything concrete.

### CODE REVIEW

**Review by yifanyang — APPROVED** [review]
Thanks for looking into it!

File: firebase-storage/src/test/java/com/google/firebase/storage/UploadTest.java [hunk 1]
```diff
@@ -128,56 +128,38 @@ public void fileUploadWith500() throws Exception {
...
-  //      Assert.fail();
-  //    } catch (IllegalStateException ignore) {
-  //      // Task is not yet done.
-  //    }
-  //
-  //    Assert.assertNull(task.getException());
-  //
-  //    task.addOnFailureListener(
-  //        (exception) -> {
-  //          Assert.assertEquals(
```
Comment (maneesht) [inline firebase-storage/src/test/java/com/google/firebase/storage/UploadTest.java:154]: I think it's worth adding back the message check, as the exception thrown could be something else entirely, and this test would erroneously pass

**Review by maneesht — CHANGES_REQUESTED** [review]

File: firebase-storage/src/test/java/com/google/firebase/storage/UploadTest.java (hunk 1 above)
Comment (daymxn) [reply firebase-storage/src/test/java/com/google/firebase/storage/UploadTest.java:154]: Ahh I was under the impression this was the only likely exception to be thrown. SGTM, will add an alternative check at the end- that's slightly less strict to be safe.

**Review by maneesht — APPROVED** [review]
LGTM, thanks!

**Pull request merged**